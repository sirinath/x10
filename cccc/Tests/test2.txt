# IWIR case
0 0 w a 1
1 0 w b 1
2 0 r a
2 1 r b
3 0 r b
3 1 r a

0.0 -> 2.0
1.0 -> 3.0
2.1 -> 1.0
3.1 -> 0.0
