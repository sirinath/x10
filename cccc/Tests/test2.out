0 0 w a 1
1 0 w b 1
2 0 r a  
2 1 r b  
3 0 r b  
3 1 r a  
# Manually specified dependencies
0.0 -> 2.0
1.0 -> 3.0
2.1 -> 1.0
3.1 -> 0.0
# Program-order dependencies
2.0 -> 2.1
3.0 -> 3.1
# Propagated dependencies
0.0 -> 2.1
1.0 -> 3.1
2.0 -> 1.0
3.0 -> 0.0
Happens-before dependencies for task 0
Happens-before dependencies for task 1
Happens-before dependencies for task 2
0.0 -> 2.0
0.0 -> 2.1
2.0 -> 1.0
2.0 -> 2.1
2.1 -> 1.0
Happens-before dependencies for task 3
1.0 -> 3.0
1.0 -> 3.1
3.0 -> 0.0
3.0 -> 3.1
3.1 -> 0.0
No cycles found: plausible CCCC execution
